### ISSUE TITLE
torch.meshgrid has no docstring if typing.TYPE_CHECKING is True

### ISSUE THREAD

**Issue by vishwakftw** [pytorch/pytorch#53545]
## 🐛 Bug

When `typing.TYPE_CHECKING` is `True`, `torch.meshgrid` doesn't have a docstring. While using IPython or the Python interpreter, this consequently reveals no docstring.

This could be the problem.

https://github.com/pytorch/pytorch/blob/3403babd940c41dee8d4f341d6befd243d3ab087/torch/functional.py#L412-L418

## Expected behavior

I feel PyTorch functions should have docstrings regardless.

## Environment

 - PyTorch Version (e.g., 1.0): 1.7.0
 - OS (e.g., Linux): Ubuntu 18.04
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source): N/A
 - Python version: 3.7

cc @brianjo @mruberry @ezyang @malfet @rgommers @xuzhao9 @gramster @gmagogsfm

**Comment by rgommers** [comment]
> When `typing.TYPE_CHECKING` is `False`, `torch.meshgrid` doesn't have a docstring.

Did you mean `True` rather than `False`?

I don't see a problem here. `TYPE_CHECKING` is only `True` when running static type checkers. At which point there's no need for a docstring.

**Comment by vishwakftw** [comment]
Oh, sorry. I did mean `True`. I specifically found this bug while using the Python interpreter or IPython, and it becomes inconvenient to have to look up the documentation via the docs site.

**Comment by rgommers** [comment]
In IPython things work fine for me with pytorch master, both with `torch.meshgrid?` and:
```
In [7]: torch.meshgrid.__doc__
Out[7]: 'Take :math:`N` tensors, each of which can be either scalar or 1-dimensional\n
```
I tried in IPython 6.5.0, this may be an IPython regression (`readline` support was broken for some `jedi` version). Does `torch.meshgrid.__doc__` contain the right content for you?

**Comment by rgommers** [comment]
> (and on the standard Python interpreter) returns `None`.

This means that it's not a regression.

You're using PyTorch 1.7.0, for which `meshgrid` simply does not have a docstring: https://github.com/pytorch/pytorch/blob/1.7/torch/functional.py#L352

This is fixed in 1.8.0 and in master (see https://github.com/pytorch/pytorch/blob/master/torch/functional.py#L417), so I'll close the issue.
